source/server/listener_impl.cc

```diff
@@ -536,5 +622,85 @@ void ListenerImpl::setSocketFactory(const Network::ListenSocketFactorySharedPtr&
   socket_factory_ = socket_factory;
 }
 
+bool ListenerImpl::supportUpdateFilterChain(const envoy::config::listener::v3::Listener& config,
+                                            bool worker_started) {
+  if (!Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.listener_in_place_filterchain_update")) {
+    return false;
+  }
+
+  // If a listener is in active list but stopped, the in place update will fail. The only known case
+  // is that listener is stopped by admin interface. It is ok if that admin method is called only
+  // prior to stop envoy. Delegate to full listener update for the least user astonishment.
+  // TODO(lambdai): Figure out if we should warn and ignore the update in both in place update and
+  // full listener update.
+  if (is_stopped_) {
+    return false;
```

Comment:
Did you add a test for this case? I didn't see it in the diff. Please check the coverage report to make sure all of these false cases have test coverage.

Reply:
No... Adding now

Reply:
I am running test and I find the case is handled already. Will revert my duplicated flags